### ISSUE TITLE
Drag & Drop not possible from Sidebar

### ISSUE THREAD

**Issue by jansende** [linuxmint/nemo#2984]
```
 * nemo 5.2.4
 * desktop
 * Mint 20.3 Cinnamon
 * 64 bit
 ```

**Issue**
Drag & Drop does not work from Sidebar.
![drag drop 2](https://user-images.githubusercontent.com/44576417/167844411-7c5a0bf8-e2ea-44b7-a71c-716143870cd2.jpg)

**Steps to reproduce**
1. Open two Nemo Windows.
2. Select a file/folder in the Main Window..
3. Try Draging&Dropping the file/folder from one Nemo Window into the other. (It works.)
4. Open the Sidebar.
5. Select a folder in the Sidebar.
3. Try Draging&Dropping the file/folder from one Nemo Window into the other. (It does not work.)

**Expected behaviour**
Drag & Drop from the Sidebar should work as if the selected folder was dragged from the Main Window.

**Comment by Jeremy7701** [comment]
The sidebar does not contain files; it contains bookmarks.

**Comment by jansende** [comment]
Please expand. I do not understand.

**Comment by Jeremy7701** [comment]
I think I may have been mistaken in identifying the 'red' area as a sidebar (press F9 to actually get/remove a sidebar).

On inspection, you appear to have one window, split into two unequal panes, with the red pane showing your /home/username directory and the green pane showing /home/username/music/itunes.

Do you have write authority to the **itunes** directory? If not, you should get an error message when trying to drag a file to it.

**Comment by MANICX100** [comment]
I second as an important defect to fix.
That the 'folders; are infact bookmarks  shouldn't matter the same way on windows if I pin a folder to Quick Access. The item in quick access is a shortcut but it is smart enough to take any files dragged into it as move this here.

**Comment by MANICX100** [comment]
> I believe there is some confusion here:- @jansende 'sidebar' - note that it appears to contain the entire home directory. ![Screenshot from 2022-05-12 15-40-32](https://user-images.githubusercontent.com/52585119/168102417-b73ca941-0e97-4d75-890a-28ae90c53d0f.png)
> 
> However a true sidebar - which can be toggled with F9, does allow dragging a file/folder into the sidebar. It is necessary to choose whether the purpose of the drag is to create a link, move or copy the file/folder. The ALT key allows this to be done with a prompt.
> 
> Have to say I didn't previously know this was possible.
> 
> My toolbar looks like this: ![Screenshot from 2022-05-12 15-43-26](https://user-images.githubusercontent.com/52585119/168103724-7e9cd2cd-fe81-48b0-b08d-e3661dc7638a.png)

Ah I understand.

@jansende 
So the 'folders' you see under my computer don't work as folders do you can't drag into them they are strictly shortcuts.
To drag and drop you must first setup folders under the bookmarks sections where you can drag/drag these then work as traditional folders to drag/drop.

@Jeremy7701 -
I would counter that this is still very unintuitive, especially as if I press Ctrl+B for bookmarks I see both real 'bookmarks' and shortcut 'bookmarks' in this case.
Expected behavior would replicate windows as this would save me having to setup true bookmarks.

**Comment by Jeremy7701** [comment]
> @Jeremy7701 -
I would counter that this is still very unintuitive, especially as if I press Ctrl+B for bookmarks I see both real 'bookmarks' and shortcut 'bookmarks' in this case.

Ctrl+B should only show `nemo` bookmarks.

By 'real bookmarks' do you mean symbolic links in the style of ![Screenshot from 2022-05-12 17-55-50](https://user-images.githubusercontent.com/52585119/168128743-8ac0b0be-9df2-4159-851f-0806fc5e1947.png) ?

**Comment by MANICX100** [comment]
> > @Jeremy7701 -
> > I would counter that this is still very unintuitive, especially as if I press Ctrl+B for bookmarks I see both real 'bookmarks' and shortcut 'bookmarks' in this case.
> 
> Ctrl+B should only show `nemo` bookmarks.
> 
> By 'real bookmarks' do you mean symbolic links in the style of ![Screenshot from 2022-05-12 17-55-50](https://user-images.githubusercontent.com/52585119/168128743-8ac0b0be-9df2-4159-851f-0806fc5e1947.png) ?

Ctrl+B shows all both types of bookmarks. By this I mean that listed under My Computer and last listed under bookmarks for me.

@Jeremy7701 -
You can wordaround this issue for now by going Bookmarks, Edit Bookmarks and deleting all.
Then go into each folder wanted for drag and drop and add them via Bookmarks, Add bookmark

You can confirm the folders have been added by looking in
`sudo nano ~/.config/gtk-3.0/bookmarks`

(nano not required)

You will then be able to drag and drop once relaunching nemo with folders on the left.

**Comment by Jeremy7701** [comment]
I have no problem.
`cat ~/.config/gtk-3.0/bookmarks` reflects my nemo bookmarks.
I did **not** delete anything!

**Comment by MANICX100** [comment]
> I have no problem. `cat ~/.config/gtk-3.0/bookmarks` reflects my nemo bookmarks. I did **not** delete anything!

Sorry I @'d  the wrong person.

**Comment by smurphos** [comment]
There's a bit of confusion in this thread - nemo's sidebar can operate in Places (i.e default places + Bookmarks) view or Treeview (which is what the OP's original screenshot showed). The sidebar in either view does not display files only folders, and the sidebar in either view a is valid _destination_ for drag and drop, but not a valid source (it's a bit pointless having it as a source when you could only move folders).

OP - you may benefit from dual pane mode dedicating one pane in list view for your wanted workflow.

**Comment by jansende** [comment]
Wow! That generated much more response than I expected :open_mouth: 

@Jeremy7701 
> I think I may have been mistaken in identifying the 'red' area as a sidebar (press F9 to actually get/remove a sidebar).
> 
> On inspection, you appear to have one window, split into two unequal panes, with the red pane showing your /home/username directory and the green pane showing /home/username/music/itunes.
> 
> Do you have write authority to the **itunes** directory? If not, you should get an error message when trying to drag a file to it.

My screenshot shows the sidebar you are talking about. This is not a split window. F9 toggles it.

I have write permission, but that is not the problem. I do not know how to create a screenshot of this, but when I drag from the main window, the mouse cursor changes. If I do it from the sidebar, nothing happens.

**Comment by jansende** [comment]
@MANICX100 
> I second as an important defect to fix. That the 'folders; are infact bookmarks shouldn't matter the same way on windows if I pin a folder to Quick Access. The item in quick access is a shortcut but it is smart enough to take any files dragged into it as move this inside.

I agree. From an UX perspective that makes the most sense.

**Comment by jansende** [comment]
@Jeremy7701 
> However a true sidebar - which can be toggled with F9, does allow dragging a file/folder into the sidebar. It is necessary to choose whether the purpose of the drag is to create a link, move or copy the file/folder. The ALT key allows this to be done with a prompt.

You are right. It seems like it is possible to drag files INTO the sidebar. However, it is not possible to drag them FROM the sidebar.

> My toolbar looks like this: ![Screenshot from 2022-05-12 15-43-26](https://user-images.githubusercontent.com/52585119/168103724-7e9cd2cd-fe81-48b0-b08d-e3661dc7638a.png)

For me it is the same. However, I can toggle between bookmarks, and the whole filesystem with the buttons at the bottom left.

![grafik](https://user-images.githubusercontent.com/44576417/168458511-eb36a44f-a9ed-4c51-af6a-87d124c79167.png)

**Comment by jansende** [comment]
@MANICX100 
> @jansende So the 'folders' you see under my computer don't work as folders do you can't drag into them they are strictly shortcuts. To drag and drop you must first setup folders under the bookmarks sections where you can drag/drag these then work as traditional folders to drag/drop.

Although I think this is weird from a user perspective (_Why are bookmarks different to folders?_), I could accept this as an exception. However, I still think it has to work when using the tree view of the filesystem as it looks like a normal window view and thus normal window actions are expected to work.

**Comment by jansende** [comment]
@smurphos 
> There's a bit of confusion in this thread - nemo's sidebar can operate in Places (i.e default places + Bookmarks) view or Treeview (which is what the OP's original screenshot showed). The sidebar in either view does not display files only folders, and the sidebar in either view a is valid _destination_ for drag and drop, but not a valid source (it's a bit pointless having it as a source when you could only move folders).

Thanks for bringing that to a point.

My problem is exactly that. If I have normal window open, I can move folders via drag and drop from the tree. However, I cannot move them using the sidebar. This is weird!

> OP - you may benefit from dual pane mode dedicating one pane in list view for your wanted workflow

I personally dislike splitting the window. In my workflow I am constantly opening and closing windows, snapping them to quadrants of my two screens. All of this is not supported by a split window...
